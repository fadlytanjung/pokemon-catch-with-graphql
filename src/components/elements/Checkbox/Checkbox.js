import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Checkbox(props) {
  const { className, onChange, checked, disabled } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');
  return <input checked={checked} className={customClass} disabled={disabled} onChange={onChange} type="checkbox" />;
}

Checkbox.defaultProps = {
  checked: false,
  className: '',
  disabled: false,
  onChange: () => {},
};

Checkbox.propTypes = {
  checked: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
};
