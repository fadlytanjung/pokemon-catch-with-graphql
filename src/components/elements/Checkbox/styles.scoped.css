.root {
  appearance: none;
  background-color: #ffffff;
  border: 0.0625rem solid #cccccc;
  border-radius: 0.25rem;
  height: 1.25rem;
  position: relative;
  transition: background-color 0.15s;
  width: 1.25rem;
}
.root:checked {
  background-color: var(--primary-color);
  border: none;
}
.root:checked::after {
  color: #ffffff;
  content: '\2713';
  line-height: 1.25rem;
  position: absolute;
  text-align: center;
  vertical-align: middle;
  width: 100%;
}
