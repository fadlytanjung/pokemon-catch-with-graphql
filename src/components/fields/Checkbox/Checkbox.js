import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function CheckBox(props) {
  const { className, input, inputProps, meta } = props;
  const { dirty, error, touched } = meta;
  const classes = [
    styles.root,
    error &&  (dirty || touched) && styles.error,
    className,
  ].filter(Boolean).join(' ');

  return (
    <>
      <label className={styles.label}>
        <input {...inputProps} {...input} className={classes} type="checkbox"/>
        <span>{inputProps.label}</span>
      </label>
      {error &&  (dirty || touched) && <small className={styles.error}>{error}</small>}
    </>
  );
}

CheckBox.defaultProps = {
  className: '',
  input: {},
  inputProps: {},
  meta: {}
};

CheckBox.propTypes = {
  className: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  meta: PropTypes.object
};
