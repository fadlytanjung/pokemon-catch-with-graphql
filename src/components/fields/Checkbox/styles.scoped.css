.root {
  appearance: none;
  background-color: #ffffff;
  border: 0.0625rem solid #cccccc;
  border-radius: 0.25rem;
  cursor: pointer;
  height: 1.25rem;
  margin-right: 1.25rem;
  position: relative;
  transition: background-color 0.15s;
  width: 1.25rem;
}

.root:checked {
  background-color: #EB5757;
  border: none;
}

.root:checked::after {
  content: '\2713';
  color: #ffffff;
  line-height: 1.25rem;
  position: absolute;
  text-align: center;
  vertical-align: middle;
  width: 100%;
}


.root.error {
  border-color: #ee4c24;
}

.root + small.error {
  color: #ee4c24;
  display: block;
  font-size: 0.75rem;
  margin-top: 0.25rem;
}

.label {
  align-items: center;
  color: #EB5757;
  display: flex;
  font-weight: 400;
  font-size: 1rem;
  text-transform: uppercase;
  letter-spacing: 0.0625rem;
}
