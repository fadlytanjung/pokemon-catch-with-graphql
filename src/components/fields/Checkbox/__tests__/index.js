import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Checkbox from '../Checkbox';

describe('src/components/fields/Checkbox', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Checkbox />);
    expect(tree).toMatchSnapshot();

    const result = Checkbox({
      ...Checkbox.defaultProps,
      input: { value: 'test' },
      inputProps: { label: 'abc' },
    });
    const label = result.props.children[0].props.children[1];
    expect(label).toEqual(<span>abc</span>);
  });

  test('render error', () => {
    const meta = { error: 'error', touched: true };
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Checkbox meta={meta} />);
    expect(tree).toMatchSnapshot();
  });
});
