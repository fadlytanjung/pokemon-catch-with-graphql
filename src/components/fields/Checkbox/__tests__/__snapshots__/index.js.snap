// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Checkbox render 1`] = `
<React.Fragment>
  <label>
    <input
      className=""
      type="checkbox"
    />
    <span />
  </label>
</React.Fragment>
`;

exports[`src/components/fields/Checkbox render error 1`] = `
<React.Fragment>
  <label>
    <input
      className=""
      type="checkbox"
    />
    <span />
  </label>
  <small>
    error
  </small>
</React.Fragment>
`;
